testutils/src/main/java/com/amplifyframework/testutils/sync/SynchronousAuth.java

```diff
@@ -252,11 +254,52 @@ public Object confirmResetPassword(
      */
     @NonNull
     public AuthSession fetchAuthSession() throws AuthException {
-        return Await.<AuthSession, AuthException>result(AUTH_OPERATION_TIMEOUT_MS, (onResult, onError) ->
-                asyncDelegate.fetchAuthSession(onResult, onError)
+        return Await.result(AUTH_OPERATION_TIMEOUT_MS, asyncDelegate::fetchAuthSession);
+    }
+
+    /**
+     * Remembers current device synchronously.
+     * @return Dummy object - just indicates it completed successfully
+     * @throws AuthException exception
+     */
+    public Object rememberDevice() throws AuthException {
+        return Await.<Object, AuthException>result(AUTH_OPERATION_TIMEOUT_MS, (onResult, onError) ->
+                asyncDelegate.rememberDevice(() -> onResult.accept(new Object()), onError)
+        );
+    }
+
+    /**
+     * Forgets the current device synchronously.
+     * @return Dummy object - just indicates it completed successfully
+     * @throws AuthException exception
+     */
+    public Object forgetDevice() throws AuthException {
+        return Await.<Object, AuthException>result(AUTH_OPERATION_TIMEOUT_MS, (onResult, onError) ->
+                asyncDelegate.forgetDevice(() -> onResult.accept(new Object()), onError)
+        );
+    }
+
+    /**
+     * Forgets the current device synchronously.
+     * @param device Auth device to forget
+     * @return Dummy object - just indicates it completed successfully
+     * @throws AuthException exception
+     */
+    public Object forgetDevice(@NonNull AuthDevice device) throws AuthException {
+        return Await.<Object, AuthException>result(AUTH_OPERATION_TIMEOUT_MS, (onResult, onError) ->
+                asyncDelegate.forgetDevice(device, () -> onResult.accept(new Object()), onError)
         );
     }
```

Comment:
Passing the dummy `new Object()` into the `Await` util is okay. But you don't need to return it.

Success can be == doesn't throw an exception.

I'd update this for every method that uses this pattern.

```suggestion
    /**
     * Forgets the current device.
     * @param device Auth device to forget
     * @throws AuthException On failure to forget the provided device
     */
    public void forgetDevice(@NonNull AuthDevice device) throws AuthException {
        Await.<Object, AuthException>result(AUTH_OPERATION_TIMEOUT_MS, (onResult, onError) ->
                asyncDelegate.forgetDevice(device, () -> onResult.accept(new Object()), onError)
        );
    }
```

Reply:
This is the pattern that I usually follow, but I was trying to match the other methods in this class.